src/DatePicker/Calendar.js

```diff
@@ -42,6 +42,7 @@ class Calendar extends Component {
     onTouchTapOk: PropTypes.func,
     open: PropTypes.bool,
     shouldDisableDate: PropTypes.func,
+    showDateDisplay: PropTypes.bool,
```

Comment:
I have some consideration for the naming convention and the API consistency. With boolean properties, we have two options. `showX`/`hideX` I think that we should be choosing based on the default value. That's basically what have been done at the HTML specification, e.g. `disabled`, `readonly` in order to allow the shorthand.
Let's follow the same convention. with `hideX` as the default is going to be `false`.

TL;DR What about `hideCalendarDate`?